lesson12/index.js

```diff
@@ -0,0 +1,92 @@
+const ul = document.getElementById('js-list');
+const button = document.getElementById('js-button');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.style.listStyle = "none";
+    li.id = "js-loading";
+    loadingImg.src = 'loading-circle.gif';
+    ul.appendChild(li).appendChild(loadingImg);
+};
+
+const removeLoading = () => document.getElementById('js-loading').remove();
+
+const removeButton = () => document.getElementById('js-button').remove();
+
+const createErrorMessage = (error) => {
+    const li = document.createElement("li");
+    li.textContent =`エラー内容：${error.message}`;
+    ul.appendChild(li);
+    console.error(error.message);
+};
+
+/**
+ * Async関数 
+ * @return {Array} 
+ */
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/5tqv';
+const fetchData = async () => {
+    try {
+        const response = await fetch(jsonUrl) // 中身はresponseオブジェクトを含むpromise
+        const json = await response.json(); //response オブジェクトから JSON を抽出
+        return json.data;
+
+    } catch(error) {
+        throw new Error(error);
+    } 
+};
+
+/**
+ * Async関数
+ * fetchData関数から値が返ってきた後、その値を返す
+ */
+const fetchListData = async () => {
+    createLoading();
+    try {
+        const data = await fetchData();
```

Comment:
recommend
```suggestion
        const data = await fetchData(jsonUrl);
```

Reply:
Thank you for review!
Changed the json value passing part.
[d4987c0](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/12/commits/d4987c0ab6c2551da3747e587dd9c66dafd66fef)